segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/MemoryStateUpdater.java

```diff
@@ -135,15 +135,30 @@ void process(Operation operation) throws DataCorruptionException {
             addToReadIndex((StorageOperation) operation);
             if (operation instanceof StreamSegmentAppendOperation) {
                 // Transform a StreamSegmentAppendOperation into its corresponding Cached version.
+                StreamSegmentAppendOperation appendOp = (StreamSegmentAppendOperation) operation;
                 try {
-                    operation = new CachedStreamSegmentAppendOperation((StreamSegmentAppendOperation) operation);
+                    operation = new CachedStreamSegmentAppendOperation(appendOp);
                 } catch (Throwable ex) {
                     if (Exceptions.mustRethrow(ex)) {
                         throw ex;
                     } else {
                         throw new DataCorruptionException(String.format("Unable to create a CachedStreamSegmentAppendOperation from operation '%s'.", operation), ex);
                     }
                 }
+
+                // Release the memory occupied by this StreamSegmentAppendOperation's BufferView - it has been processed
+                // and is no longer needed.
+                try {
+                    appendOp.close();
+                } catch (Throwable ex) {
```

Comment:
This method doesn't actually have a throws in it's signature.

Reply:
Neither does `ByteBuf.release`, but it does throw an `IllegalStateException` if you release an already freed object. The point of this method is to clearly log cases (and label them appropriately) if we cannot release the memory we held for this class.